src/Microsoft.VisualStudio.Services.Agent/ProcessChannel.cs

```diff
@@ -50,15 +50,18 @@ public sealed class ProcessChannel : AgentService, IProcessChannel
         private StreamString _writeStream;
         private StreamString _readStream;
 
-        public void StartServer(StartProcessDelegate startProcess)
+        public void StartServer(StartProcessDelegate startProcess, bool disposeClient = true)
```

Comment:
Nit: to be more specific, maybe this is `disposeStreamClientsAfterStart` or something like that? It's hard to tell from current naming exactly what it's doing (and why).